### PR TITLE
Added sink rule for Third party SDK (firebase)

### PR DESCRIPTION
**Pull request by hiaashna**
- `react-firebase-hooks`: Third-party library for React that provides hooks for Firebase Realtime Database and Firebase Authentication, so it may involve the flow of personal data if the data stored or used by the hooks includes personal information.

- `Firebase-admin`: It allows the developer to manage user accounts and authenticate with Firebase using custom authentication tokens or service accounts, so it also may involve the flow of personal data like email, password and name.

- `Firebase Authentication`: It allows users to sign in to their app using their email address and password, or using their Google, Facebook, or Apple account. Personal information such as the user's email address, name, and profile picture may be collected and stored.

- `Firebase Realtime Database`: It allows developers to store and sync data in real-time. Data stored includes personal information, such as user profiles or messages.

- `Firebase Cloud Firestore`: It allows developers to store and retrieve data in a NoSQL document-based database. Like the Realtime Database, Data stored includes personal information.

- `Firebase Cloud Storage`: It allows to store and retrieve files such as images, videos, and audio. If the files stored include personal information, it will be stored in the Firebase Cloud Storage.

- `Firebase Cloud Messaging`: It allows to send push notifications to users. A user's device token, which can be used to send notifications to a specific device, is collected as a personal data

### CODE REVIEW

File: rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml [hunk 1]
```diff
@@ -11,3 +11,59 @@ sinks:
     patterns:
       - "ngx-auth-firebaseui|@asithade\\/ngx-auth-firebaseui"
     tags:
+
+  - id: ThirdParties.SDK.Firebaseapp.React-firebase-hooks
```
Comment (abhstabs) [inline rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml:15]: @khemrajrathore are we ok with this naming convention? I think capitalisation works better than using hyphens. What are your thoughts?

File: rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml [hunk 2]
```diff
@@ -11,3 +11,59 @@ sinks:
...
+    patterns:
+      - "firebase-admin|@firebase/admin"
+    tags:
+
+  - id: Firebase.SDK.Authentication
+    name: Firebase Authentication
+    domains:
+      - "firebase.com", "google.com", "facebook.com", "apple.com"
+    patterns:
+      # There is no specific regex pattern for Firebase Authentication, as the SDK allows users to sign in using multiple providers, each with their own pattern.
```
Comment (abhstabs) [inline rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml:36]: @hiaashna Can you share the examples here? If we don't have the pattern the rule will not be of much use

**Review by abhstabs — CHANGES_REQUESTED** [review]
The regular expressions are not correct.

File: rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml (hunk 2 above)
Comment (hiaashna) [reply rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml:36]: I agree that without a specific pattern, the rule provided for Firebase Authentication may not be as useful in detecting personal data. The Firebase Authentication SDK allows users to sign in with multiple providers, such as Google, Facebook and each provider may have a different format for the personal data they collect.

few eg:
- Google: pattern: "(?i)("id_token":")(.*)(","access_token")"
- Facebook: pattern: "(?i)("access_token":")(.*)(","token_type")"

File: rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml (hunk 1 above)
Comment (khemrajrathore) [reply rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml:15]: Yes, I agree, having a standard naming convention is essential. Let's follow the capitalized convention

**Review by khemrajrathore — COMMENTED** [review]
@hiaashna pattern regex added are not correct. we apply these regex on package names, not API URLs or strings.

@abhstabs we need to give her more context on this

**Comment by khemrajrathore** [comment]
@hiaashna Change the base branch to ruleSegregation
cc- @abhstabs

File: rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml (hunk 1 above)
Comment (hiaashna) [reply rules/sinks/third_parties/sdk/firebaseapp/javascript.yaml:15]: ok

**Pull request closed without merging**